src/DenseACSets.jl

```diff
@@ -644,6 +652,21 @@ end
 ACSetInterface.cascading_rem_parts!(acs::ACSet, type, parts) =
   delete_subobj!(acs, Dict(type=>parts))
 
+function ACSetInterface.undefined_subparts(acs::SimpleACSet, f::Symbol)
+  s = acset_schema(acs)
+  f ∈ homs(s; just_names=true) || error("$f is not a hom")
+  _undefined_subparts(acs, f, parts_type(acs))
```

Comment:
I don't think `parts_type` is actually needed to be called for dispatch since you can do 
```
function undefined_subparts(acs::SimpleACSet{<:DenseParts}, f::Symbol) ... end 

function undefined_subparts(acs::SimpleACSet{<:MarkAsDeleted}, f::Symbol) ... end 
``` 

If there's not much shared code between the two implementations you don't need a top level `undefined_parts` that calls `_undefined_parts`. (In particular, I think checking that `f` is in `homs(s)` is implicitly done by indexing `acs.subparts[f]` in each of the implementations.)